src/vs/workbench/contrib/inlineChat/browser/inlineChatDecorations.ts

```diff
@@ -125,14 +125,26 @@ export class InlineChatDecorationsContribution extends Disposable implements IEd
 		decorationUpdateScheduler.schedule();
 	}
 
-	private _onSelectionOrContentChanged(editor: IActiveCodeEditor): void {
+	private async _onSelectionOrContentChanged(editor: IActiveCodeEditor): Promise<void> {
 		const selection = editor.getSelection();
+		const selectionIsEmpty = selection.isEmpty();
 		const model = editor.getModel();
 		const uri = model.uri;
+
 		const isInlineChatVisible = this._inlineChatSessionService.getSession(editor, uri);
 		const startLineNumber = selection.startLineNumber;
 		const hasBreakpoint = this._debugService.getModel().getBreakpoints({ uri: uri, lineNumber: startLineNumber }).length > 0;
-		const isEnabled = selection.isEmpty() && /^\s*$/g.test(model.getLineContent(startLineNumber)) && !isInlineChatVisible && !hasBreakpoint;
+
+		const startPosition = selection.getStartPosition();
+		const endPosition = selection.getEndPosition();
+		const startWord = this._editor.getConfiguredWordAtPosition(startPosition);
+		const endWord = this._editor.getConfiguredWordAtPosition(endPosition);
+		const shouldBeVisibleOnNonEmptySelection = startWord && endWord && startPosition.column <= startWord.startColumn && endPosition.column >= endWord.endColumn;
```

Comment:
Does this test if the selection is at the start/end of a word? Will this work if I select from bracket to bracket?
Maybe all we need to test is that we are not in the middle of a word.

Reply:
This tests that the word at the beginning of the selection and at the end of the selection is fully covered. Meaning that if the selection is as follows (where the pipe (|) is the selection start and end):

const someWor|d = 'h|i';

then this will be not fully covered, however the following is:

const |someWord = 'hi|';

This logic does not work when you select bracket to bracket because when the cursor is at a bracket, the method getWordAtPosition returns null, since there is no word. Now that you mention it, we could indeed test that we are not in the middle of the word. I'll do that.